src/gossip/service.zig

```diff
@@ -1856,62 +1888,102 @@ pub const GossipService = struct {
         return nodes[0..node_index];
     }
 
-    /// Frees heap allocated memory of filtered values
+    /// Sorts the incoming `gossip_values` slice to place the valid gossip data
+    /// at the start, and returns the number of valid gossip values in that slice.
     pub fn filterBasedOnShredVersion(
         self: *Self,
         gossip_table: *const GossipTable,
         gossip_values: []SignedGossipData,
-        from_pubkey: Pubkey,
+        sender_pubkey: Pubkey,
     ) usize {
-        // we use swap remove which just reorders the array
-        // (order dm), so we just track the new len -- ie, no allocations/frees
-        const my_shred_version = self.my_shred_version.load(.monotonic);
-        if (my_shred_version == 0) {
-            return gossip_values.len;
-        }
+        const S = struct {
```

Comment:
would ideally want this logic to be as simple as possible unless we need it faster - can we return to how it was originally wrote? - running this on testnet and the table gets zero values inserted (guessing somethings going wrong here)

Reply:
interesting. I am quite against putting back the old logic since its time complexity is large for no reason. can you instead share how you tested it and I'll just fix the issue?

Reply:
makes sense - time complexity is important but only when the code is a hot path and needs optimization (not convinced this code needs it) - our first priority is simplicity and performance second which is why i prefer the original approach

would prefer to keep this pr changes restricted to the channel too - happy to discuss further in a separate pr

run it on testnet and youll see values arent inserted

Reply:
> makes sense - time complexity is important but only when the code is a hot path and needs optimization (not convinced this code needs it) - our first priority is simplicity and performance second which is why i prefer the original approach

yeah good point. i will revert it and look into it later.

Reply:
done: 45915bc69ff3f713167694db02a03127d7dc4e83